provider/azure/azure_privatedns_test.go

```diff
@@ -354,9 +354,9 @@ func TestAzurePrivateDNSApplyChanges(t *testing.T) {
 }
 
 func TestAzurePrivateDNSApplyChangesDryRun(t *testing.T) {
-	recordsClient := mockRecordSetsClient{}
+	recordsClient := mockPrivateRecordSetsClient{}
 
-	testAzureApplyChangesInternal(t, true, &recordsClient)
+	testAzurePrivateDNSApplyChangesInternal(t, true, &recordsClient)
```

Comment:
While implementing my change I noticed the private zone tests were using an `AzureProvider` instead of a `AzurePrivateDNSProvider`